proxy/http/HttpSM.cc

```diff
@@ -1760,6 +1764,10 @@ HttpSM::state_http_server_open(int event, void *data)
 
     session->new_connection(vc);
 
+#ifdef HAVE_SYSTEMTAP
+    DTRACE_PROBE1(trafficserver, new_origin_server_connection, t_state.current.server->name);
+#endif
```

Comment:
Can't we create ts_sdt.h with something like:
```
#pragma once

#ifdef HAVE_SYSTEMTAP

#include <sys/sdt.h>

#define DTRACE_PROBE DTRACE_PROBE1

#else 

#define DTRACE_PROBE(...)

#endif

```

Reply:
There are multiple DTRACE_PROBE macros, one for each accepted number of arguments.

For example, DTRACE_PROBE1 provides 1 argument, DTRACE_PROBE2 provides 2, and so forth.

Reply:
So then instead:
```
#pragma once

#ifdef HAVE_SYSTEMTAP

#include <sys/sdt.h>

#else 

#define DTRACE_PROBE1(...)
#define DTRACE_PROBE2(...)
#define DTRACE_PROBE3(...)
// and so forth

#endif